ext/standard/dl.c

```diff
@@ -205,6 +205,11 @@ PHPAPI int php_load_extension(const char *filename, int type, int start_now)
 		return FAILURE;
 	}
 	module_entry = get_module();
+	if (zend_hash_str_exists(&module_registry, module_entry->name, strlen(module_entry->name))) {
+		DL_UNLOAD(handle);
```

Comment:
Nit: even though the module isn't deallocated here it might be more intuitive to move the unload call after the error to make it obvious this isn't accessing freed memory.

Reply:
Hmm, that is handled pretty inconsistently in this function; sometimes the error is raised before, and sometimes after the unload. I'm not sure if it's worth to unify all occurrences.

Reply:
I don't mind either way